* Process Samples [/]
  This is all done in the no_region_removal folder
** TODO NGSCat QC
   mkdir ngscat
   ln -s ~/scratch/resequencing/no_region_removal/*recal_reads.bam


** DONE Single Sample Pre-Processing
*** DONE Samtools filtering
   #This is the Script that will rerun the GATK analysis for the resequence_data
   #Uses some of the GATK best practices

Plus follows as closely as possible the GATK best practices
http://www.broadinstitute.org/gatk/guide/best-practices?bpm=DNAseq#best-practices-ovw
parallel -j 1   "samtools view -bu {} | samtools sort -n -o - samtools_nsort_tmp | \
           samtools fixmate /dev/stdin /dev/stdout | samtools sort -o - samtools_csort_tmp |\
           samtools fillmd -u - ../references/GRCh37-lite.fa > {/.}.fixed.bam" ::: *.bam
*** DONE RealignTargetCreator
   parallel -j 8 "java -Xmx16g -jar ../GenomeAnalysisTK.jar -T  RealignerTargetCreator -R ../references/GRCh37-lite.fa -I {} -o {/.}.intervals --known ../references/ALL.wgs.indels_mills_devine_hg19_leftAligned_collapsed_double_hit.indels.sites.vcf.gz --known ../references/ALL.wgs.low_coverage_vqsr.20101123.indels.sites.vcf.gz" ::: *.fixed.bam
*** Done IndelRealigner
    parallel -j 8 "java -Xmx16g -jar ../GenomeAnalysisTK.jar -T  IndelRealigner -R ../references/GRCh37-lite.fa -I {} --targetIntervals {/.}.intervals -o {/.}.realigned_indels.bam -known ../references/ALL.wgs.indels_mills_devine_hg19_leftAligned_collapsed_double_hit.indels.sites.vcf.gz -known ../references/ALL.wgs.low_coverage_vqsr.20101123.indels.sites.vcf.gz" ::: *.fixed.bam
*** DONE BASE Recalibration
   parallel -j 8 "java -Djava.io.tmpdir=/Volumes/BiochemXsan/scratch/merrimanlab/james/resequencing/bams/tmp -Xmx32g -jar ../GenomeAnalysisTK.jar -T BaseRecalibrator -R ../references/GRCh37-lite.fa -I {}  -o {/.}.table -knownSites ../references/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz -knownSites ../references/ALL.wgs.indels_mills_devine_hg19_leftAligned_collapsed_double_hit.indels.sites.vcf.gz  -knownSites ../references/ALL.wgs.low_coverage_vqsr.20101123.indels.sites.vcf.gz -log {/.}.base_recal.log " ::: *.fixed.realigned_indels.bam
    parallel -j 8 "java -Djava.io.tmpdir=/Volumes/BiochemXsan/scratch/merrimanlab/james/resequencing/bams/tmp -Xmx20g -jar ../GenomeAnalysisTK.jar -T BaseRecalibrator -R ../references/GRCh37-lite.fa -I {} -L  -o {/.}.post.table -BQSR {/.}.table -knownSites ../references/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz -knownSites ../references/ALL.wgs.indels_mills_devine_hg19_leftAligned_collapsed_double_hit.indels.sites.vcf.gz -knownSites ../references/ALL.wgs.low_coverage_vqsr.20101123.indels.sites.vcf.gz -log {/.}.second_base_recal.log " ::: *.fixed.realigned_indels.bam
*** DONE Plot Generation
   parallel -j 8 "java -Djava.io.tmpdir=/Volumes/BiochemXsan/scratch/merrimanlab/james/resequencing/bams/tmp -Xmx32g -jar ../GenomeAnalysisTK.jar -T AnalyzeCovariates -R ../references/GRCh37-lite.fa  -L  -plots {/.}.pdf -before {/.}.table -after {/.}.post.table -log {/.}.plots " ::: *.realigned_indels.table
*** DONE PrintReads
    parallel -j 8 "java -Djava.io.tmpdir=/Volumes/BiochemXsan/scratch/merrimanlab/james/resequencing/bams/tmp -Xmx32g -jar ../GenomeAnalysisTK.jar -T PrintReads -R ../references/GRCh37-lite.fa -I {}  -BQSR {/.}.table -o {/.}.recal_reads.bam" ::: *.fixed.realigned_indels.bam
** TODO Variant Calling [%]

*** DONE HaplotypeCaller
parallel -j 8 "java -Djava.io.tmpdir=/Volumes/BiochemXsan/scratch/merrimanlab/james/resequencing/bams/tmp -Xmx32g -jar ../GenomeAnalysisTK.jar -T HaplotypeCaller -R ../references/GRCh37-lite.fa  --dbsnp ../references/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz -I {} -o {/.}.gvcf --emitRefConfidence GVCF --variant_index_type LINEAR --variant_index_parameter 128000" ::: *.recal_reads.bam
*** TODO Genotype GVCF
    #Full VCF
    java -Djava.io.tmpdir=/Volumes/BiochemXsan/scratch/merrimanlab/james/resequencing/bams/tmp -Xmx32g -jar ../GenomeAnalysisTK.jar -T GenotypeGVCFs  -R ../references/GRCh37-lite.fa  --dbsnp ../references/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz  --max_alternate_alleles 20 \
`for gvcf in *.gvcf
do
    echo "--variant ${gvcf}"
done` \
-o resequencing_raw.vcf --includeNonVariantSites
  # Subset VCF 
    java -Djava.io.tmpdir=/Volumes/BiochemXsan/scratch/merrimanlab/james/resequencing/bams/tmp -Xmx32g -jar ../GenomeAnalysisTK.jar -T GenotypeGVCFs  -R ../references/GRCh37-lite.fa  --dbsnp ../references/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz  --max_alternate_alleles 20 \
`for gvcf in *.gvcf
do
    echo "--variant ${gvcf}"
done` \
-o resequencing_smaller.vcf -L  regions_gatk.bed --includeNonVariantSites
  # Finally only the regions we asked for
    java -Djava.io.tmpdir=/Volumes/BiochemXsan/scratch/merrimanlab/james/resequencing/bams/tmp -Xmx32g -jar ../GenomeAnalysisTK.jar -T GenotypeGVCFs  -R ../references/GRCh37-lite.fa  --dbsnp ../references/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz  --max_alternate_alleles 20   \
`for gvcf in *.gvcf
do
    echo "--variant ${gvcf}"
done` \
-o resequencing_smaller_asked_for.vcf -L sorted_regions.bed --includeNonVariantSites

*** TODO Unified Genotyper

java -Djava.io.tmpdir=/Volumes/BiochemXsan/scratch/merrimanlab/james/resequencing/bams/tmp -Xmx200g -jar ../GenomeAnalysisTK.jar -nt 4 -T  UnifiedGenotyper -R ../references/GRCh37-lite.fa  --dbsnp ../references/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz  \
`for name in *.recal_reads.bam
do
    echo "-I ${name}"
done` \
-o raw_ug.vcf \
--output_mode EMIT_ALL_SITES
** TODO Variant Score Quality recalibration
      Refer to the documentation in he VQSR folder for information about this.

