* VQSR
** DONE SNP RECAL
*** Build Model
java -jar ../GenomeAnalysisTK.jar \ 
    -T VariantRecalibrator \ 
    -R ../references/GRCH37/human_g1k_v37_decoy.fasta \ 
    -input output_raw_vcf.vcf \
    -resource:hapmap,known=false,training=true,truth=true,prior=15.0 hapmap.vcf \ 
    -resource:omni,known=false,training=true,truth=true,prior=12.0 omni.vcf \ 
    -resource:1000G,known=false,training=true,truth=false,prior=10.0 1000G.vcf \ 
    -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 dbsnp.vcf \ 
    -an QD \ 
    -an FS \ 
    -an MQRankSum \ 
    -an ReadPosRankSum \ 
    -mode SNP \ 
    -nt 6 \
    -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \ 
    -recalFile recalibrate_SNP.recal \ 
    -tranchesFile recalibrate_SNP.tranches \ 
    -rscriptFile recalibrate_SNP_plots.R 
*** Apply Recalibration.
     java -jar ../GenomeAnalysisTK.jar \
    -T ApplyRecalibration \
    -R ../references/GRCH37/human_g1k_v37_decoy.fasta \
    -input output_raw_vcf.vcf \
    -mode SNP \
    --ts_filter_level 99.0 \ 
    -recalFile recalibrate_SNP.recal \
    -tranchesFile recalibrate_SNP.tranches \
    -o recalibrated_snps_raw.vcf \
    -nt 6 
** TODO INDEL RECAL (Didn't Work)
*** Generate Tranches
      java -jar ../GenomeAnalysisTK.jar \
    -T VariantRecalibrator \
    -R ../references/GRCH37/human_g1k_v37_decoy.fasta \
    -input recalibrated_snps_raw_indels.vcf \
    -resource:mills,known=true,training=true,truth=true,prior=12.0 mill.vcf \
    -an DP \
    -an FS \
    -an MQRankSum \
    -an ReadPosRankSum \
    -mode INDEL \
    -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
    --maxGaussians 4 \
    -recalFile recalibrate_INDEL.recal \
    -tranchesFile recalibrate_INDEL.tranches \
    -rscriptFile recalibrate_INDEL_plots.R 
*** Apply Recalibration
    java -jar ../GenomeAnalysisTK.jar \ 
    -T ApplyRecalibration \ 
    -R ../references/GRCH37/human_g1k_v37_decoy.fasta \ 
    -input recalibrated_snps_raw_indels.vcf \ 
    -mode INDEL \ 
    --ts_filter_level 99.0 \ 
    -recalFile recalibrate_INDEL.recal \ 
    -tranchesFile recalibrate_INDEL.tranches \ 
    -o recalibrated_variants.vcf  
** DONE HARD FILTER INDELS (Too small call set)
*** Generate indels (raw) and snps(VQSR).
java -jar ../GenomeAnalysisTK.jar \ 
    -T SelectVariants \ 
    -R ../references/GRCH37/human_g1k_v37_decoy.fasta  \
    -V recalibrated_snps_raw.vcf \
    -selectType INDEL \ 
    -o raw_indels.vcf 

java -jar ../GenomeAnalysisTK.jar \
    -T SelectVariants \
    -R ../references/GRCH37/human_g1k_v37_decoy.fasta  \
    -V recalibrated_snps_raw.vcf \
    -selectType SNP \
    -o vqsr_snp.vcf 

*** Filter INDELS
java -jar ../GenomeAnalysisTK.jar \ 
    -T VariantFiltration \ 
    -R ../references/GRCH37/human_g1k_v37_decoy.fasta\ 
    -V raw_indels.vcf \ 
    --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" \ 
    --filterName "my_indel_filter" \ 
    -o filtered_indels.vcf 

*** Rename SampleNames
    # change the sample names
      bcftools reheader --samples sample_replace.txt vqsr_snp.vcf > \
    vqsr_snp_names.vcf
      bcftools reheader --samples sample_replace.txt filtered_indels.vcf > \
    filtered_indels_names.vcf
*** Subset out populations for PSEQ_RUN
    # PWD
    /Volumes/BiochemXsan/staff_users/jamesboocock_temp/scratch/resequencing/pipeline
    # EURO filters 
    bcftools view -S poly_samples.txt -b total_regions.bed
    total_regions.bed | \
    bcftools filter -e "AC = 0" > \
    vcf_final/poly_snps.vcf
      bcftools view -S euro_samples.txt vqsr_snp_names.vcf -b  \
    total_regions.txt | \
      bcftools filter -e "AC = 0" > \
    vcf_final/euro_snps.vcf

    bcftools view -S poly_samples.txt filtered_indels_names.vcf | \
      bcftools filter -e "AC = 0" > \
    vcf_final/poly_indels.vcf

    bcftools view -S euro_samples.txt filtered_indels_names.vcf | \
      bcftools filter -e "AC = 0" > \
    vcf_final/euro_indels.vcf
*** Subset out populations from the smaller VCF
    bedtools intersect -header -a vqsr_snp_names.vcf -b sorted_regions.bed > \
    snps_bed_subset.vcf
    bcftools view -S poly_samples.txt snps_bed_subset.vcf | \
     bcftools filter -e "AC = 0"    > \
    vcf_final_smallest/poly_snps.vcf
    bcftools view -S poly_samples.txt filtered_indels_names.vcf | \
      bcftools filter -e "AC = 0" > \
    vcf_final_smallest/poly_indels.vcf
    
    # Blah
    bedtools intersect -header -a filtered_indels_names.vcf -b total_regions_new.bed > \
    snps_bed_subset.vcf
    bcftools view -S poly_samples.txt snps_bed_subset.vcf | \

    bcftools view -S euro_samples.txt filtered_indels_names.vcf > \
      bcftools filter -e "AC = 0" > \
    vcf_final/euro_indels.vcf
      bcftools view -S euro_samples.txt snps_bed_subset.vcf | \
      bcftools filter -e "AC = 0" > \
    vcf_final_smallest/euro_snps.vcf
    bedtools intersect -header -a vqsr_snp_names.vcf -b total_regions_new.bed > \
    test_subset.vcf
    bedtools intersect -header -a filtered_indels_names.vcf -b \
    total_regions_new.bed > vcf_indel_subset.vcf
    bcftools view -S poly_samples.txt snps_bed_subset.vcf | \
      bcftools filter -e "AC = 0"    > \
    vcf_final_small/poly_snps.vcf
      bcftools view -S euro_samples.txt snps_bed_subset.vcf | \
      bcftools filter -e "AC = 0" > \
    vcf_final_small/euro_snps.vcf
    bcftools view -S poly_samples.txt vcf_indel_subset.vcf | \
     bcftools filter -e "AC = 0"    > \
    vcf_final_small/poly_indels.vcf
    bcftools view -S euro_samples.txt vcf_indel_subset.vcf | \
      bcftools filter -e "AC = 0" > \
    vcf_final_small/euro_indels.vcf

      # TODO intels

FINISHED 25/10/2014 4:04PM
*** Extract DHS peaks
    bcftools view -S poly_samples.txt -R bed_files/dhs.bed \
    vqsr_snp_names.vcf.gz | bcftools filter -e "AC = 0" > \
    dhs/poly_snps.vcf
*** Extract Exons + promoter
    bcftools view -S poly_samples.txt -R bed_files/exon_sorted.bed \
    vqsr_snp_names.vcf.gz | bcftools filter -e "AC = 0" > \
    only_genes/poly_snps.vcf

*** :w

